pool/src/lib.rs

```diff
@@ -430,18 +580,22 @@ impl LendingPoolTrait for LendingPool {
         amount: i128,
         balance_from_before: i128,
         balance_to_before: i128,
-        s_token_supply: i128,
+        total_collat: i128,
     ) -> Result<(), Error> {
         // TODO: maybe check with callstack?
         require_not_paused(&env);
 
         let reserve = read_reserve(&env, &asset)?;
         require_active_reserve(&env, &reserve);
-        require_zero_debt(&env, &to, &reserve.debt_token_address);
+
+        let debt_token = DebtTokenClient::new(&env, &reserve.debt_token_address);
+        let to_debt = debt_token.balance(&to);
+        require_zero_debt(&env, to_debt);
 
         reserve.s_token_address.require_auth();
 
-        let reserve = recalculate_reserve_data(&env, &asset, &reserve, s_token_supply)?;
+        let total_debt = debt_token.total_supply();
+        let reserve = recalculate_reserve_data(&env, &asset, &reserve, total_collat, total_debt)?;
```

Comment:
We can call read_reserve here as reserve now recalculates after deposit/repay/borrwo ops